pkg/configuration/configuration.go

```diff
@@ -0,0 +1,63 @@
+package configuration
+
+import (
+	"fmt"
+	"os"
+)
+
+type ErrorCollector struct {
+	Missing []string
+}
+
+func (e *ErrorCollector) Register(name string) {
+	e.Missing = append(e.Missing, name)
+}
+
+func (e *ErrorCollector) Error() error {
+	if len(e.Missing) == 0 {
+		return nil
+	} else {
+		var message string
+		for _,name := range e.Missing {
+			message += fmt.Sprintf("env variable with name %s was not found, ", name)
+		}
+		return fmt.Errorf(message)
+	}
+}
+
+type Configuration struct {
+	GoogleCredentials string
+	RedshiftHostTemplate string
+	RedshiftUsername string
+	RedshiftPassword string
+	RedshiftMasterDatabase string
+	AwsAccountId string
+	Region string
+	GoogleAdminPrincipalEmail string
+}
+
+func loadVariable(name string, errorCollector *ErrorCollector) string {
+	if value, ok := os.LookupEnv(name); ok {
+		return value
+	} else {
+		errorCollector.Register(name)
+		return ""
+	}
```

Comment:
Just a note on happy path optimization. Usually we try to keep the happy path left most and error handling indented.
This could be written like this instead.

```suggestion
	value, ok := os.LookupEnv(name)
	if !ok {
		errorCollector.Register(name)
		return ""
	}
	return value
```